Feature/update filter output record

ChrysmOre opened the pull request:
### What

When CSV export is triggered by a filter job, update filter output record with download link. Otherwise update instance.

### How to review

Check changes make sense, tests pass

### Who can review

Anyone

PauloPortugal commented:
The [rebase strategy](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) is advisable on private branches as avoids having to have a develop merge commit there. Also I would avoid "Fix xxx" as it does not provide any benefit and is essentially work of a previous commit, that now has tests against it.

I don't block PRs because of it, but it also removes noise from the `git log` where I don't need to see _"Fix x"_, _"addressed PR comments"_ or  _"Forgot to do Y"_. This gives a cleaner main branch and cleaner logs and is still part of the same logical piece of work, so we could amend a commit instead.

I probably just care too much about commits. :popcorn: :sweat_smile:

PauloPortugal commented on handler/handler_test.go:
```diff
@@ -518,6 +529,50 @@ func TestGetS3ContentLength(t *testing.T) {
+				So(err, ShouldBeNil)
+			})
+
+			Convey("Then the expected UpdateInstance call is executed with the expected paramters", func() {
```
probably best to replace `Then` with `And`
Non blocker.

ChrysmOre commented:
> The [rebase strategy](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) is advisable on private branches as avoids having to have a develop merge commit there. Also I would avoid "Fix xxx" as it does not provide any benefit and is essentially work of a previous commit, that now has tests against it.
> 
> I don't block PRs because of it, but it also removes noise from the `git log` where I don't need to see _"Fix x"_, _"addressed PR comments"_ or _"Forgot to do Y"_. This gives a cleaner main branch and cleaner logs and is still part of the same logical piece of work, so we could amend a commit instead.
> 
> I probably just care too much about commits. 🍿 😅

Yeah whether or not I rebase usually depends on how complicated the merge errors are likely to be/how lazy I'm feeling :)
